File: CS/RemoveItemsFromPopupMenu/Form1.cs
using System.Windows.Forms;
using DevExpress.XtraPdfViewer;
using DevExpress.XtraBars;
using DevExpress.XtraPdfViewer.Extensions;
using DevExpress.XtraPdfViewer.Commands;

namespace CustomPopupMenu
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            pdfViewer1.LoadDocument("..\\..\\Demo.pdf");
        }

        void pdfViewer1_PopupMenuShowing(object sender, PdfPopupMenuShowingEventArgs e)
        {

            // Create a bar button item.
            BarButtonItem browseBarButton = new BarButtonItem();
            browseBarButton.Caption = "Custom Item";

            // Insert the bar buttom item into the PDF Viewer popup menu and start a new group.
            e.ItemLinks.Add(browseBarButton, true);

            // Handle the bar button click event.
            browseBarButton.ItemClick += browseBarButton_ItemClick;

            // Hide the popup menu for the bookmark tree.
            if (e.PopupMenuKind == PdfPopupMenuKind.BookmarkTree)
            {
                e.ItemLinks.Clear();
            }

            // Remove Rotate Clockwise and Rotate Counterclockwise items from the Page Content popup menu.
            if (e.PopupMenuKind == PdfPopupMenuKind.PageContent)
            {
                e.ItemLinks.Remove(e.ItemLinks.GetPdfViewerBarItemLink(PdfViewerCommandId.RotatePageClockwise));
                e.ItemLinks.Remove(e.ItemLinks.GetPdfViewerBarItemLink(PdfViewerCommandId.RotatePageCounterclockwise));
            }
        }

        void browseBarButton_ItemClick(object sender, ItemClickEventArgs e)
        {
            MessageBox.Show("ItemClick event fires");
        }
    }
}

File: VB/RemoveItemsFromPopupMenu/Form1.vb
Imports System.Windows.Forms
Imports DevExpress.XtraPdfViewer
Imports DevExpress.XtraBars
Imports DevExpress.XtraPdfViewer.Extensions
Imports DevExpress.XtraPdfViewer.Commands

Namespace CustomPopupMenu

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
            pdfViewer1.LoadDocument("..\..\Demo.pdf")
        End Sub

        Private Sub pdfViewer1_PopupMenuShowing(ByVal sender As Object, ByVal e As PdfPopupMenuShowingEventArgs)
            ' Create a bar button item.
            Dim browseBarButton As BarButtonItem = New BarButtonItem()
            browseBarButton.Caption = "Custom Item"
            ' Insert the bar buttom item into the PDF Viewer popup menu and start a new group.
            e.ItemLinks.Add(browseBarButton, True)
            ' Handle the bar button click event.
            AddHandler browseBarButton.ItemClick, AddressOf browseBarButton_ItemClick
            ' Hide the popup menu for the bookmark tree.
            If e.PopupMenuKind = PdfPopupMenuKind.BookmarkTree Then
                e.ItemLinks.Clear()
            End If

            ' Remove Rotate Clockwise and Rotate Counterclockwise items from the Page Content popup menu.
            If e.PopupMenuKind = PdfPopupMenuKind.PageContent Then
                e.ItemLinks.Remove(e.ItemLinks.GetPdfViewerBarItemLink(PdfViewerCommandId.RotatePageClockwise))
                e.ItemLinks.Remove(e.ItemLinks.GetPdfViewerBarItemLink(PdfViewerCommandId.RotatePageCounterclockwise))
            End If
        End Sub

        Private Sub browseBarButton_ItemClick(ByVal sender As Object, ByVal e As ItemClickEventArgs)
            MessageBox.Show("ItemClick event fires")
        End Sub
    End Class
End Namespace
